harness/determined/cli/experiment.py

```diff
@@ -1091,7 +1091,8 @@ def experiment_id_arg(help: str) -> Arg:  # noqa: A002
                     "--local",
                     action="store_true",
                     help="Create the experiment in local mode instead of submitting it to the "
-                    "cluster. For more information, see documentation on det.experimental.create()",
+                    "cluster. Requires --test.  For more information, see the How to Debug Models "
```

Comment:
We've got two separate things, `local` mode and `test` mode:
* test: actually create the experiment or just validate it
* local: run model on local machine or talk to master

Right now, `local` is only implemented for test mode. I get the desire to document that limitation in the help, but I think it's slightly dangerous to do so for when `local` can actually train models, too. Is that ever going to be a thing?

Reply:
`--local` and `--test` are considered legacy features, since by their nature they assume they imply that we own the control loop and know how to configure those things.

They're only supported for Trial class entrypoints (the legacy entrypoints).  They're not supported for pytorch.Trainer, instead you get a warning that says, "why don't you just call trainer.fit() yourself" or something.

So there's no risk in this case.

That said, I'm also happy to remove the mention of `--test` if you prefer, because the error you would get is self-explanatory.  Just let me know.  I care a lot that our documentation doesn't lie, but I don't care as much about the finer points of how we document legacy features.

Reply:
If there's no risk, I'm happy with either solution (like you, I also think the error when `--test` isn't passed is pretty good). Since you've implemented it with a description of `--test`, I take that to mean both that you like that message and that the test-inclusive message is easiest to merge. 🚢 it!